src/com/cft/contactmerge/tests/CompareContactPartsTest.java

```diff
@@ -786,30 +830,75 @@ void doPhoneNumbersMatch_No() {
+        Contact contactOne = new Contact();
+
+        contactOne.getPhone().setPhoneNumber("(520) 123-4567");
+        assertEquals(AnswerType.no, CompareContactParts.doPhoneNumbersMatch(contactOne, ""));
+    }
+
     /*******************************************************************************************************************
      *********************************************** Compare Email Tests ***********************************************
      *******************************************************************************************************************/
     @Test
     void doEmailsMatch_Yes_IgnoreSpaces() {
-        Contact contactOne = new Contact();
+        Email firstEmail = new Email();
+        Email secondEmail = new Email("jdoe@yahoo.com");
 
-        contactOne.getEmail().setEmailAddress("jdoe@yahoo.com");
-        assertEquals(AnswerType.yes, CompareContactParts.doEmailsMatch(contactOne, " jdoe@yahoo.com "));
+        firstEmail.setEmailAddress(" jdoe@yahoo.com ");
+        assertEquals(AnswerType.yes, CompareContactParts.doEmailsMatch(firstEmail, secondEmail));
     }
 
     @Test
-    void doEmailsMatch_Yes_IgnoreCase() {
-        Contact contactOne = new Contact();
+    void doEmailsMatch_No() {
+        Email firstEmail = new Email();
+        Email secondEmail = new Email("jdoe@gmail.com");
 
-        contactOne.getEmail().setEmailAddress("jdoe@yahoo.com");
-        assertEquals(AnswerType.yes, CompareContactParts.doEmailsMatch(contactOne, "JDoe@Yahoo.com"));
+        firstEmail.setEmailAddress("jdoe@yahoo.com");
+        assertEquals(AnswerType.no, CompareContactParts.doEmailsMatch(firstEmail, secondEmail));
     }
 
     @Test
-    void doEmailsMatch_No() {
-        Contact contactOne = new Contact();
+    void doEmailsMatch_No_Missing() {
+        Email firstEmail = new Email();
+        Email secondEmail = new Email("jdoe@yahoo.com");
+
+        assertEquals(AnswerType.no, CompareContactParts.doEmailsMatch(firstEmail, secondEmail));
+    }
+
+    /*******************************************************************************************************************
+     ******************************************* Compare Email Address Tests *******************************************
+     *******************************************************************************************************************/
+    @Test
+    void doEmailAddressesMatch_Yes_IgnoreSpaces() {
+        Email originalEmail = new Email();
+
+        originalEmail.setEmailAddress("jdoe@yahoo.com");
+        assertEquals(AnswerType.yes, CompareContactParts.doEmailAddressesMatch(originalEmail, " jdoe @yahoo.com "));
+    }
+
+    @Test
+    void doEmailAddressesMatch_Maybe_IgnoreCase() {
+        Email originalEmail = new Email();
+
+        originalEmail.setEmailAddress("jdoe@yahoo.com");
+        assertEquals(AnswerType.maybe, CompareContactParts.doEmailAddressesMatch(originalEmail, "JDoe@Yahoo.com"));
```

Comment:
Capitalization doesn't matter in names.

https://www.computerhope.com/issues/ch000709.htm